kurl_proxy/cmd/main.go

```diff
@@ -191,6 +191,15 @@ func getHttpServer(fingerprint string) *http.Server {
 			"fingerprintSHA1": fingerprint,
 		})
 	})
+	r.NoRoute(func(c *gin.Context) {
+		target := url.URL{
+			Scheme:   "https",
+			Host:     c.Request.Host,
+			Path:     c.Request.URL.Path,
+			RawQuery: c.Request.URL.RawQuery,
+		}
+		c.Redirect(http.StatusMovedPermanently, target.String())
```

Comment:
@divolgin are you ok with this being a 301 vs a 302?

Reply:
i just went with this here
https://github.com/unrolled/secure/blob/v1/secure.go#L321-L324

Reply:
this seems more official though. perhaps ill go with 301
https://support.google.com/webmasters/answer/6073543?hl=en

Reply:
oops, that is what i have!